dev/tasks/tasks.yml

```diff
@@ -1517,9 +1516,9 @@ tasks:
       image: conda-python-hdfs
 {% endfor %}
 
-{% for python_version, spark_version, test_pyarrow_only, numpy_version in [("3.7", "v3.1.2", "false", "latest"),
-                                                                           ("3.8", "v3.2.0", "false", "1.23"),
-                                                                           ("3.9", "master", "false", "latest")] %}
+{% for python_version, spark_version, test_pyarrow_only, numpy_version in [("3.8", "v3.1.2", "false", "latest"),
```

Comment:
Do we still need Spark 3.1.2?

I think that Spark 3.1 reached EOL:

https://spark.apache.org/versioning-policy.html

> Feature release branches will, generally, be maintained with bug fix releases for a period of 18 months.

And 3.1.1 was released on 2021-03-02:

https://spark.apache.org/news/index.html

> [Spark 3.1.1 released](https://spark.apache.org/news/spark-3-1-1-released.html)
>
> March 2, 2021

3.2 might reach EOL because 3.2.0 was released on 2021-10-13:

> [Spark 3.2.0 released](https://spark.apache.org/news/spark-3-2-0-released.html)
>
> October 13, 2021

@kiszk Do you know what Spark versions are maintained?

Reply:
@kou should we move the spark versions conversation to update them to its own ticket?

Reply:
Ah, yes. It makes sense.

Reply:
I have created this issue to track it: https://github.com/apache/arrow/issues/36199